Made the Armory Gun Safe a craftable object

TyAshley opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
Changed Gun Safes to be craftable with 10 steel, 5 LV cables, and 10 plasteel. Takes longer to construct than regular closets/lockers.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
If something breaks a safe or if new weapons are unlocked in science, it is impossible for the Warden/HoS to create a new safe to hold these weapons.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
YML only changes. 

Create entity "GunSafeBase" that has no default access, and can be constructed with 10 steel, 5 LV cable, and 10 plasteel. It has a price of 335 spesos (280 for 10 plasteel, 50 for 10 steel, and 5 for the 5 LV cables). 

Create entity "GunSafeBaseSecure" with "GunSafeBase" as a parent. This has an access level of "Armory"

Changed filled gun safes to inherit from "GunSafeBaseSecure," so that they still spawn requiring Armory access 

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.
-->
![image](https://github.com/user-attachments/assets/1736d498-0fa2-45d7-98e3-cbe9e9ffbffe)
![image](https://github.com/user-attachments/assets/06e72453-b4fb-40ea-abc9-acbeb912ed1b)
![image](https://github.com/user-attachments/assets/35bfb6f3-50e3-4976-b9f6-8fe7de6b3b86)
![image](https://github.com/user-attachments/assets/ba6d9c94-c213-446a-a19f-3ab90b683a4b)
![image](https://github.com/user-attachments/assets/31dcd3cf-88ac-476c-8fda-f7cf5e839428)
![image](https://github.com/user-attachments/assets/a16d7301-b54a-43dd-867d-c51bfea277a0)

![image](https://github.com/user-attachments/assets/2c022c09-aada-4a2a-b12c-7684733f7fa9)
![image](https://github.com/user-attachments/assets/1d6d44f4-de05-4e29-83e0-7ae0c83d18b0)
![image](https://github.com/user-attachments/assets/18a2b1cc-b7f8-493f-9fde-01fb2bf155d9)
![image](https://github.com/user-attachments/assets/e97b1af1-b890-4bad-b57f-afa7966b0c6d)

## Requirements
<!-- 
Due to influx of PR's we require to ensure that PR's are following the correct guidelines.

Please take a moment to read these if its your first time.

Check the boxes below to confirm that you have in fact seen these (put an X in the brackets, like [X]):
-->
- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->
Renamed entity "GunSafe" to "GunSafeBase" that has no default access, and can be constructed, with a cost of 335 spesos (280 for 10 plasteel, 50 for 10 steel, and 5 for the 5 LV cables). 

Create entity "GunSafeBaseSecure" with "GunSafeBase" as a parent. This has access level of "Armory"

Changed filled gun safes to inherit from "GunSafeBaseSecure" vs "GunSafe"

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->
:cl: Allen
- add: Made Gun Safes a craftable object (10 steel, 5 lv cable, 10 plasteel)
<!--
Make sure to take this Changelog template out of the comment block in order for it to show up. Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog.
:cl: Allen
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->

lzk228 commented:
it should start without accesses, so ce or hop has to configure it later

TyAshley commented:
I do see the test failing (and agree it is a valid test failure), and am taking a look at the spesos difference between materials and the safe itself. I'll also need to double-check the process for starting a constructed object with no accesses set.


> > it should start without accesses, so ce or hop has to configure it later
> 
> I agree. It is weird if you can build it, but are unable to open it. Also the test fail is real. If you deconstruct it the materials are worth more spesos than the safe itself.

lzk228 commented:
> I'll also need to double-check the process for starting a constructed object with no accesses set.

yeah, you just need to create gun safe entity without accesses and use it for construction
then entity with accesses should inherit from the first one
and all filled versions should inherit from the one with accesses

TyAshley commented:
Corrected the cost to match the cost of materials (335 total). I also updated the constructed gun safe to start with no access levels, and need to be adjusted with an access configurator. The map-spawn filled gun safes continue to have an armory access level.

lzk228 commented on Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml:
```diff
@@ -350,7 +350,7 @@
 
 - type: entity
-  id: GunSafe
+  id: GunSafeBase
```
better to not change the id
it's fine with the current

lzk228 commented on Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml:
```diff
@@ -359,6 +359,19 @@
+  - type: StaticPrice
+    price: 335
+
+- type: entity
+  id: GunSafeBaseSecure
+  parent: GunSafeBase
+  components:
```
add a suffix `Armory` or something, so in spawn menu you could see the difference

slarticodefast commented:
Are you coming back to this?
If yes, please address the review and the test fail.

slarticodefast commented:
If you are coming back to this, leave a comment and I will re-open the PR.
